src/main/java/spoon/template/StatementTemplate.java

```diff
@@ -43,8 +43,7 @@ public CtStatement apply(CtType<?> targetType) {
 		CtClass<?> c = Substitution.getTemplateCtClass(targetType, this);
 		// we substitute the first statement of method statement
 		CtStatement result = c.getMethod("statement").getBody().getStatements().get(0).clone();
-		new SubstitutionVisitor(c.getFactory(), targetType, this).scan(result);
-		return result;
+		return (CtStatement) new SubstitutionVisitor(c.getFactory(), targetType, this).substitute(result);
```

Comment:
It seems that the class Substitute also use the `SubstitutionVisitor.scan()` usage you let it as it is on purpose or is it a miss?